src/docfx/lib/schema/JsonSchemaTransformer.cs

```diff
@@ -38,6 +38,12 @@ public JsonSchemaTransformer(JsonSchema schema)
             var errors = new List<Error>();
             var xrefSpecs = new List<InternalXrefSpec>();
             LoadXrefSpecsCore(file, context, _schema, token, errors, xrefSpecs);
+
+            // if only one uid defined in the file, remove the bookmark from href if any
+            if (xrefSpecs.Count == 1)
```

Comment:
why not move line 83 here now that `href` is settable?

Reply:
Because it would be easier to identify if the UID is at the top level during the traversal.